### PR TITLE
Add a pull test job focussing on InPlacePodVerticalScaling feature with containerd-main

### PR DESCRIPTION
**Pull request by vinaykul**
This test job is intended to allow full-spectrum E2E test run for PR https://github.com/kubernetes/kubernetes/pull/102884 which is possible if we have containerd>=1.6.9 that brings CRI support for the feature.

The existing pull test job pull-kubernetes-e2e-gce-alpha-features provides multi-VM coverage with AllAlpha features enabled.

Once we have the cos OS updated with containerd >= 1.6.9, this test job can be removed.

### CODE REVIEW

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml
Comment (bobbypage) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1613]: remove this, this is not needed (it's only needed for the `kubelet-credential-provider` job)

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml
Comment (bobbypage) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1602]: you need to add 


```
KUBE_MASTER_EXTRA_METADATA / KUBE_NODE_EXTRA_METADATA
```

since we are using containerd main branch

see https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-node/containerd.yaml#L1068-L1070

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml
Comment (vinaykul) [reply config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml]: fixed both issues. Please give it an ok-to-test to check if CI likes it(local make verify looks good).

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml
Comment (bobbypage) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1618]: I don't think you should have this, it means to use the latest build of k8s, but it should be using the one from the pull job. I think this should be `--extract=local` (same as https://github.com/kubernetes/test-infra/blob/bba2146583ec442decce2a8dab2d348d25e0004e/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L356)

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml
Comment (bobbypage) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1626]: I'm pretty sure you can drop this -- I don't see it used in the `pull-kubernetes-e2e-gce-alpha-features` job

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml
Comment (vinaykul) [reply config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:1618]: i'll update it shortly

**Comment by SergeyKanzhelev** [comment]
> Once we have the cos OS updated with containerd >= 1.6.9, this test job can be removed.

Can you please put this as a comment or a job description?

Also we had AllAlpha before. Would it be best just to add AllAlpha optional job back?

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve
/hold

please add a comment or description for this job. I understand this is needed to unblock KEP so approving to get it going. Unhold once description is added

**Comment by vinaykul** [comment]
> > Once we have the cos OS updated with containerd >= 1.6.9, this test job can be removed.
> 
> Can you please put this as a comment or a job description?
> 
> Also we had AllAlpha before. Would it be best just to add AllAlpha optional job back?

Yes I'll update the PR shortly and create an issue to track this future rollback task.

**Comment by vinaykul** [comment]
@SergeyKanzhelev Added comment and tracking this AI in issue https://github.com/kubernetes/kubernetes/issues/113838

**Comment by bobbypage** [comment]
/lgtm
/unhold
(comment for the rationale of this job was added)

**Pull request merged**
